pkg/sync/isync.go

```diff
@@ -41,6 +42,10 @@ type ISync interface {
 	// Sync is the contract between Runtime and sync implementation.
 	// Note that, it is expected to return the first data sync as soon as possible to fill the store.
 	Sync(ctx context.Context, dataSync chan<- DataSync) error
+
+	// ReSync is used to validate the flag store following a delete event
```

Comment:
```suggestion
	// ReSync is used to validate the flag store following a delete event
```

I'm not sure how I might change it, but I think this function is more "generally useful" than this comment implies. It might also be useful if we moved to using deltas in gRPC and we needed to get the whole state for a brand new subscriber, for instance.